flutter_searchbox_ui/lib/src/components/range_input.dart

```diff
@@ -1016,6 +1016,21 @@ class _RangeInputInnerState extends State<RangeInputInner> {
   @override
   void initState() {
     super.initState();
+    widget.searchController.subscribeToStateChanges((changes) {
```

Comment:
@mohdashraf010897 You don't need to subscribe here. You should get the controller `value` property from `SearchWidgetConnector`.

Reply:
used below
`  @override
  void didUpdateWidget(RangeInputInner oldWidget) {
    super.didUpdateWidget(oldWidget);
    var updatedValue = widget.searchController.value;
    if (!isEqual(updatedValue, {
      "start": dropdownValues['startValue'],
      "end": dropdownValues['endValue']
    })) {
      if (updatedValue == null ||
          (updatedValue is Map && updatedValue.isEmpty)) {
        dropdownValues['startValue'] = (widget.range.start is List
            ? (widget.range.start[widget.range.start.length - 1] == 'no_limit'
                ? widget.range.start[widget.range.start.length - 1]
                : widget.range.start[0] == 'other'
                    ? widget.range.start[1]
                    : widget.range.start[0])
            : "");
        dropdownValues['endValue'] = (widget.range.end is List
            ? (widget.range.end[widget.range.end.length - 1] == 'no_limit'
                ? widget.range.end[widget.range.end.length - 1]
                : widget.range.end[0] == 'other'
                    ? widget.range.end[1]
                    : widget.range.end[0])
            : "");
      } else {
        dropdownValues['startValue'] = updatedValue['start'] ??
            (widget.range.start is List &&
                    widget.range.start[widget.range.start.length - 1] ==
                        'no_limit'
                ? widget.range.start[widget.range.start.length - 1]
                : "");
        dropdownValues['endValue'] = updatedValue['end'] ??
            (widget.range.end is List &&
                    widget.range.end[widget.range.end.length - 1] == 'no_limit'
                ? widget.range.end[widget.range.end.length - 1]
                : "");
      }
      setState(() {
        dropdownValues['defaultStartValue'] = dropdownValues['startValue'];
        dropdownValues['defaultEndValue'] = dropdownValues['endValue'];
      });
      print('updated $dropdownValues');
    }
  }
`